### ISSUE TITLE
only able to run Wmail as root

### ISSUE THREAD

**Issue by nemoskapetanos** [Thomas101/wmail#205]
Hi Thomas,

Your app is great! Makes managing multiple gmail accounts really easy!

I have one small problem. I can only run the app as root. I'm a new Ubuntu 14.04 user, so I'm not sure if the problem is with the app or my Ubuntu setup.

I can see that all the program files are owned by me and not by root. Are there any dependencies needed to run the app that could be owned by root and that need the permissions changed?

Thanks for helping me out!

Best regards,
Nemo

**Comment by Thomas101** [comment]
I don't think you need to run as root? How are you opening it up? Just double clicking?

**Comment by nemoskapetanos** [comment]
Yes exactly.I tried to run it by double clicking on the WMail file but nothing happens. The only way I have worked out to run it is to open the terminal, navigate to the directory where it's located and enter the following command: sudo ./WMail. I enter my password and then it runs without any problems.
What do you think is the problem? Does something need to be configured on my system?

**Comment by Thomas101** [comment]
Hmmm, I'm not sure where the problem lies with this one. I've tested on ubuntu 16.04 and linux mint 17.1 and didn't have any problems there. What about if you go to the terminal and just type `./WMail` does that launch?

**Comment by amig** [comment]
I'm using Ubuntu 14.04 and run it as user. Maybe you've installed it as root?

**Comment by nemoskapetanos** [comment]
I tried running your suggestion of launching the program by typing just ./WMail and nothing happens. I tried that the first time, then figured out that I needed to run it as sudo for it to work properly.
  
Amig,no, I haven't installed it, I just unzipped the zip file into my downloads folder and run it from there. How do you install it in Ubuntu?

**Comment by Thomas101** [comment]
You should just be able to run it from wherever you unzip it. There's a text editor called [atom](https://atom.io/) by github. This is based on the same platform as WMail. Could you try downloading that and see if it launches okay? This might point us as to whether it's to do with your configuration or WMail

**Comment by nemoskapetanos** [comment]
I downloaded atom. When I launched it, it ran without any problems and then when I closed it, there was an error message that the program had an unexpected error. I went through the error message (you can't copy the text) and it said "atom crashed with SIGSEGV in V8::Internal::GlobalHandles::Note::Release".

At the end of the message, there was something like the following comment: The following dependencies are not up to date, please update and that should stop this problem reoccurring. It listed the following dependencies: libavahi-client3, libavahi-common-data and libavahi-common3.
I updated these dependencies through the terminal. Now atom closes without an error message, but WMail still doesn't launch just by double clicking on it. I still need to launch it through the terminal with the sudo command. I've downloaded the newest version of WMail (1.3.5) and it's the same deal. 

I think it's definitely something to do with my configuration. Do you know what it could possibly be?

**Comment by nemoskapetanos** [comment]
Problem solved! The following files were set to be owned by root. I changed the ownership to my user name and everything works fine now!
~/.config/wmail
~/.config/wmail"
~/.local/share/wmail
~/.local/share/wmail/mailboxes_db.json
~/.local/share/wmail/app_db.jsonb
~/.local/share/wmail/mailboxes_db.jsonb
~/.local/share/wmail/settings_db.jsonb
~/.local/share/wmail/app_db.json
~/.local/share/wmail/settings_db.json

Since I'm a newbie to ubuntu, I really have no idea why these files in my user directory were changed to root.  Did I do something incorrectly? 

Thank you very much for all your help!

**Comment by Thomas101** [comment]
Ah good find! The only thing I can think of is if running it the first time there was a different issue, but running it with sudo the second didn't have the first issue, but by then all the files had been created as root which got us stuck in the sudo problem.

I'll have a look to see if I can reproduce it that way and put a guard in or something.

Thanks!!

**Comment by nemoskapetanos** [comment]
You're most definitely welcome! The reason I looked at the ownership issue was that my firefox had a similar problem yesterday. I googled it and came across a thread that said that the problem could be that some firefox files had their ownership changed to root. I'm new to linux/ubuntu, so I really have no idea why this happens. But I followed the instructions in the thread and afterwards firefox worked without any problems. So I just applied the same logic to WMail and it worked!
